### PR TITLE
Remove `line-height` from typography classes

### PR DESCRIPTION
**Pull request by FlyersPh9**
## Changes

- Removed `line-height` defined on typography classes, relying on #1270.

## Testing

- Changed text in CSS demo `typography.html` to lorem ipsum to more easily see line height changes when screen is narrow. <s>May revert later.</s> Reverted.

## Docs

N/A

### CODE REVIEW

File: packages/itwinui-css/backstop/results/bitmaps_reference/iTwinUI_input_Type_SearchBox_dark_0_demo-search-box_0_.png
Comment (rohan-kadkol) [inline packages/itwinui-css/backstop/results/bitmaps_reference/iTwinUI_input_Type_SearchBox_dark_0_demo-search-box_0_.png:1]: Has the "0/3" text moved up and thus is now uncentered vertically?

File: packages/itwinui-css/backstop/results/bitmaps_reference/iTwinUI_surface_Type_default_dark_0_demo-default_0_.png
Comment (rohan-kadkol) [inline packages/itwinui-css/backstop/results/bitmaps_reference/iTwinUI_surface_Type_default_dark_0_demo-default_0_.png:1]: Has the first Surface's header text become un-centered vertically?

File: packages/itwinui-css/backstop/results/bitmaps_reference/iTwinUI_tag_State_hover__focus_dark_0_demo-default_0_desktop.png
Comment (rohan-kadkol) [inline packages/itwinui-css/backstop/results/bitmaps_reference/iTwinUI_tag_State_hover__focus_dark_0_demo-default_0_desktop.png:1]: Has the text moved down and thus become un-centered vertically?

File: packages/itwinui-css/backstop/results/bitmaps_reference/iTwinUI_workflow-diagram_Type_workflow_light_0_demo-workflow_0_.png
Comment (rohan-kadkol) [inline packages/itwinui-css/backstop/results/bitmaps_reference/iTwinUI_workflow-diagram_Type_workflow_light_0_demo-workflow_0_.png:1]: As discussed, it seems that the height of the steps in the workflow diagram has decreased. I'm not sure if this was intended/ideal.

Comment (FlyersPh9) [reply packages/itwinui-css/backstop/results/bitmaps_reference/iTwinUI_workflow-diagram_Type_workflow_light_0_demo-workflow_0_.png:1]: Adding vertical padding to make the workflow diagram look like it did prior in https://github.com/iTwin/iTwinUI/pull/1619/commits/60c0f45558bd76c1d9aa211385c03b2e5ced9895.

File: packages/itwinui-css/backstop/results/bitmaps_reference/iTwinUI_surface_Type_default_dark_0_demo-default_0_.png
Comment (FlyersPh9) [reply packages/itwinui-css/backstop/results/bitmaps_reference/iTwinUI_surface_Type_default_dark_0_demo-default_0_.png:1]: Some CSS in the demo page was causing this issue.  However, with the new line height, it makes the surface headers without a button in them appear too close to the text.  Added a `min-block-size` to the surface header.  Fix in https://github.com/iTwin/iTwinUI/pull/1619/commits/c6d03364f4c8037c0fa186d76449ff0fbdad478d.

File: packages/itwinui-css/src/workflow-diagram/workflow-diagram.scss [hunk 1]
```diff
@@ -55,7 +55,7 @@
 }
 
 .iui-workflow-diagram-content {
-  padding-inline: var(--iui-size-m);
+  padding: var(--iui-size-3xs) var(--iui-size-m);
```
Comment (mayank99) [inline packages/itwinui-css/src/workflow-diagram/workflow-diagram.scss:58]:
```suggestion
  padding-block: var(--iui-size-3xs);
  padding-inline: var(--iui-size-m);
```

**Review by mayank99 — COMMENTED** [review]
This PR is currently blocking v3 release.

File: packages/itwinui-css/backstop/results/bitmaps_reference/iTwinUI_tag_State_hover__focus_dark_0_demo-default_0_desktop.png
Comment (FlyersPh9) [reply packages/itwinui-css/backstop/results/bitmaps_reference/iTwinUI_tag_State_hover__focus_dark_0_demo-default_0_desktop.png:1]: Set line height to what it was previously just to get correct alignment in https://github.com/iTwin/iTwinUI/pull/1619/commits/2cbb9b3585052d7fd65505553589468fea171b8f.

File: packages/itwinui-css/backstop/results/bitmaps_reference/iTwinUI_input_Type_SearchBox_dark_0_demo-search-box_0_.png
Comment (FlyersPh9) [reply packages/itwinui-css/backstop/results/bitmaps_reference/iTwinUI_input_Type_SearchBox_dark_0_demo-search-box_0_.png:1]: Set line height to what it was previously just to get correct alignment in https://github.com/iTwin/iTwinUI/pull/1619/commits/0cf7d7722b7cd3af5fad71ccff52efff7f01d2b0.

File: apps/storybook/cypress-visual-screenshots/baseline/Stepper.test.ts-Basic.png
Comment (rohan-kadkol) [inline apps/storybook/cypress-visual-screenshots/baseline/Stepper.test.ts-Basic.png:1]: nit: Are the step circles and step text too close to each other?

**Review by rohan-kadkol — APPROVED** [review]
There are quite a few places where the text shifted. So I first thought the text was uncentered. But after inspecting the page for a few examples, I believe they are centered indeed. I also could not find any significant issue with any components' images.

File: apps/storybook/cypress-visual-screenshots/baseline/Stepper.test.ts-Basic.png
Comment (FlyersPh9) [reply apps/storybook/cypress-visual-screenshots/baseline/Stepper.test.ts-Basic.png:1]: Added a flex gap to give it a little bit of breathing room.

**Pull request merged**